contracts/core/SedaCoreV1.sol

```diff
@@ -65,11 +79,32 @@ contract SedaCoreV1 is ISedaCore, RequestHandlerBase, ResultHandlerBase, UUPSUpg
     /// @dev Overrides the base implementation to also add the request ID and timestamp to storage
     function postRequest(
         SedaDataTypes.RequestInputs calldata inputs
-    ) public override(RequestHandlerBase, IRequestHandler) returns (bytes32) {
+    ) public payable override(RequestHandlerBase, IRequestHandler) returns (bytes32) {
+        return postRequest(inputs, 0, 0, 0);
+    }
+
+    function postRequest(
+        SedaDataTypes.RequestInputs calldata inputs,
+        uint256 requestFee,
+        uint256 resultFee,
+        uint256 batchFee
+    ) public payable override returns (bytes32) {
+        // Check that amount is equal to sum of fees
+        if (msg.value != requestFee + resultFee + batchFee) {
+            revert InvalidFeeAmount();
+        }
+
         bytes32 requestId = super.postRequest(inputs);
         _addRequest(requestId);
-        // Store the request timestamp
-        _storageV1().requestTimestamps[requestId] = block.timestamp;
+        // Store the request details
```

Comment:
Example: it's quite obvious we are storing the request details, so why are we doing it would be a better comment.